src/next/azext_next/custom.py

```diff
@@ -307,15 +363,20 @@ def _execute_recommend_commands(cmd, rec):
 
 
 def _execute_recommend_scenarios(cmd, rec):
-    for nx_cmd in rec["nextCommandSet"]:
+    exec_idx = rec.get("executeIndex")
+    for idx, nx_cmd in enumerate(rec["nextCommandSet"]):
+        cmd_active = exec_idx is None or idx in exec_idx
```

Comment:
> exec_idx is None

We have filtered on the server side, why does this happen?

Reply:
> > exec_idx is None
> 
> We have filtered on the server side, why does this happen?

The scenario recommendations by knowledge base are based on the first command of scenario. In that case, there is no field `executeIndex`.